### PR TITLE
Update main doc file

### PR DESCRIPTION
**Pull request by alex-ju**
Update documentation
- [x] Update main documentation file to reflect current architecture.
- [x] Update main readme file to explain how to import, build and use JavaScript from `govuk-frontend`

Links for reviewers:
- main documentation file [/docs/index.md](https://github.com/alphagov/govuk-frontend/blob/46c084f41c96ff87ea5b97607ddc11a8ce350f07/docs/index.md)
- main readme file [README.md](https://github.com/alphagov/govuk-frontend/blob/16489413b9548da57599a0c6267897d305dd0be9/README.md)

Trello card:
- https://trello.com/c/qSxJW8Z4

### CODE REVIEW

**Comment by alex-ju** [comment]
This is ready for review. The Browserify, Require.js ant potentially other bundlers.

File: README.md [hunk 1]
```diff
@@ -95,6 +95,64 @@ After resolving the import paths you can import GOV.UK Frontend by using:
...
+Note: The import syntax you should use depends on the JavaScript module format used by your bundler. For example, if it is using `CommonJS`, use
+
+```JS
+require('@govuk-frontend/all/all')
+```
+
+#### Polyfills
+A JavaScript polyfill provides functionality on older browsers or assistive technology that do not natively support it.
+
+The polyfills provided with GOV.UK Frontend aim to fix usability and accessibility issues. If there is a JavaScript included in the component directory, it is important to import and initialise it in your project to ensure that all users can properly use the component (see [Import Javscript](#import-javascript)).
```
Comment (igloosi) [inline README.md]: "If there is a JavaScript **file**? included in the component directory"

**Comment by igloosi** [comment]
package-log in examples/webpack?

File: examples/webpack/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        use: {
+          loader: 'babel-loader',
+          options: {
+            presets: ['env', 'es3']
+          }
+        }
+      }
+    ]
+  },
+  output: {
```
Comment (nickcolley) [inline examples/webpack/webpack.config.js]: Did you manage to avoid polyfilling the wrapper?

**Comment by alex-ju** [comment]
@igloosi yeah, we probably don't need the lock in examples

File: examples/webpack/webpack.config.js (hunk 2 above)
Comment (alex-ju) [reply examples/webpack/webpack.config.js]: yep

Comment (alex-ju) [reply examples/webpack/webpack.config.js]: will add setup for scss and images too, but may be a good idea to wait for until after the restructure happens.

**Comment by alex-ju** [comment]
@nickcolley: updated `.gitignore`.
Because the javascript for the existing components (button and details) is being automatically initialised we don't have further details on Option2. As you mentioned this can be reviewed as part of #660

**Pull request merged**